resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala

```diff
@@ -268,8 +268,13 @@ private[spark] abstract class YarnSchedulerBackend(
         logWarning(reason.toString)
         driverEndpoint.ask[Boolean](r).onFailure {
```

Comment:
I wonder if using `ask` here is doing anything useful at all. Instead, using `send` would be cheaper, and the handler for `RemoveExecutor` can log whatever errors it runs into instead.

Then this could just be:

```
if (!stopped.get()) {
  driverEndpoint.send(...)
}
```

(And the driver endpoint will probably need a minor change too.)

Reply:
Thanks @vanzin for the review.

I think it is a good suggestion, there are other places where the `RemoveExecutor` message is sending using the `ask`, are you suggesting to change those as well?

> (And the driver endpoint will probably need a minor change too.)

You mean moving the case `RemoveExecutor(executorId, reason)` from `receiveAndReply` to `receive`?

Reply:
I think when you use `send`, all the related things as you mentioned above should be changes.

Reply:
IIRC there are two different `RemoveExecutor` messages in the code, as confusing as that may be.

But if this one is used in multiple places then it's probably not worth changing right now, unless you're up for verifying the return value is not needed in the other places.

Reply:
I see there are multiple places this message is being used, but all of them are just logging for the failure. I am thinking this logging may be useful to diagnose the failures in some cases.

Reply:
Can you point out to a couple of those? I only see this `RemoveExecutor` being handled in `CoarseGrainedSchedulerBackend.scala`.

Reply:
I see these places other than `CoarseGrainedSchedulerBackend.scala` and the one present in the PR.
https://github.com/apache/spark/blob/1e82335413bc2384073ead0d6d581c862036d0f5/core/src/main/scala/org/apache/spark/executor/CoarseGrainedExecutorBackend.scala#L169
https://github.com/apache/spark/blob/6735433cde44b3430fb44edfff58ef8149c66c13/resource-managers/yarn/src/main/scala/org/apache/spark/scheduler/cluster/YarnSchedulerBackend.scala#L250
https://github.com/apache/spark/blob/a2db5c5761b0c72babe48b79859d3b208ee8e9f6/resource-managers/yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala#L653

Reply:
Yeah, as you said, there's only logging going on. There's only two things that can happen:

- the handler throws an exception instead of replying
- the RPC layer hits an error

The first can be logged at the handler (it not already logged by the RPC layer). The second is already logged by the RPC layer.

Reply:
It sounds fine, will update the PR with the changes from `ask` to `send` for `RemoveExecutor`.